### ISSUE TITLE
Hyperparameter configuration for the result in the paper

### ISSUE THREAD

**Issue by YaqianZhang** [JiangpengHe/CL-LoRA#2]
Hi there,

Cool paper and thanks a lot for making the code public. It is really helpful.

I run the code with the imgr.json and achieves the final accuracy much lower than Table 1 (74.51%). I was wondering if any key hyperparameters (e.g. learning rate) need to be adjusted.

Many thanks!

**Comment by JiangpengHe** [comment]
Hi,

Thanks for your interest!

The results reported in the paper are the average performance over 10 independent runs using grid search for hyperparameter tuning. However, the current version of the code uses fixed hyperparameters and should still achieve comparable performance.

For your reference, here are the results from a recent run using the current code:

2025-05-30 05:23:10,305 [trainer.py] => CNN top1 curve: [99.31, 91.8, 93.44, 91.29, 91.34, 89.48, 88.86, 88.49, 87.25, 86.57, 85.86, 85.21, 82.92, 83.33, 82.47, 81.91, 81.77, 80.95, 79.55, 78.83, 79.25, 78.89, 77.15, 77.83, 77.63, 77.37, 76.33, 77.48, 77.16, 76.56, 76.25, 76.76, 76.61, 76.15, 75.94, 74.82, 74.45, 74.69, 74.03, 74.55]
2025-05-30 05:23:10,305 [trainer.py] => CNN top5 curve: [100.0, 99.68, 99.65, 98.55, 98.19, 97.78, 97.89, 97.58, 97.07, 96.3, 95.85, 95.95, 95.22, 95.15, 94.89, 94.92, 94.71, 94.44, 93.81, 93.84, 93.85, 93.42, 93.22, 93.25, 92.72, 92.54, 92.48, 92.62, 92.21, 91.84, 91.97, 91.81, 91.59, 91.05, 91.27, 90.46, 90.41, 90.52, 90.23, 90.07]

Average Accuracy (CNN): 81.51325000000001
2025-05-30 05:23:10,305 [trainer.py] => Average Accuracy (CNN): 81.51325000000001, with the last step accuracy 74.55 and average 81.51%. 

Please let me know for any further questions. 

Thank you.

**Comment by YaqianZhang** [comment]
Hi,

Thanks for the reply. I was able to reproduce a similar result. The issue was the batch size. I had used 128 instead of 32 (as indicated in the imgr.json). Apologies for the confusion.

Kind regards,
Yaqian

**Comment by JiangpengHe** [comment]
Thanks for the followup, the batch size can indeed have a noticeable impact on performance. In our experiments, we found that many hyperparameters (e.g., learning rate, weight decay, and number of epochs) interact closely with the batch size, so they generally need to be tuned together rather than in isolation. In addition, hardware (e.g., GPU memory and precision settings) can also influence the optimal batch size, since it affects gradient accumulation and numerical stability. So even small differences in setup can sometimes lead to variations in the final accuracy.

Please feel free to reach out if there are any other questions.
